packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+			.attr("width", "100%")
+			.attr("height", Configuration.toolBar.height)
+			.attr("opacity", 1);
+
+		// clean children first
+		container.html(null);
+		// loading or empty state
+		if (isDataLoading) {
+			// put an empty rect to keep space unchanged
+			DOMUtils.appendOrSelect(container, "svg.toolbar-loading-space")
+				.append("rect")
+				.attr("height", Configuration.toolBar.height)
+				.attr("width", iconSize * 3) // value doesn't matter but can't be empty
+				.attr("opacity", 0);
+		} else {
+			const self = this;
+
+			const buttonList = [];
+			// add zoom in/out button only if zoom bar is enabled
+			if (this.isZoomBarEnabled) {
+				buttonList.push(this.getZoomInButtonConfig());
+				buttonList.push(this.getZoomOutButtonConfig());
+			}
+			buttonList.push(this.getOverFlowMenuButtonConfig());
+
+			// render buttons sequentially
+			let buttonXPosition = 0;
+			buttonList.forEach((button) => {
+				// zoom in icon and event
+				const buttonContainer = DOMUtils.appendOrSelect(
+					container,
+					`svg.${button.id}`
+				);
+				// add icon back ground rect
+				let buttonSVG = `<rect class="icon-${button.iconId}" x="${buttonXPosition}px" y="0px"
+					width="${iconSize}px" height="${iconSize}px"/>`;
+				buttonSVG += button.iconSVG(buttonXPosition);
+				buttonContainer.html(buttonSVG);
+				buttonContainer.on("click", button.clickFunction);
+				buttonXPosition += iconSize;
+			});
+
+			this.overflowMenuIcon = DOMUtils.appendOrSelect(
+				this.getContainerSVG(),
+				"rect.icon-overflowRect"
+			);
+
+			const hasOpenedOverflowMenuOptions =
+				this.overflowMenuOptions
+					.selectAll("ul.bx--overflow-menu-options--open")
+					.size() > 0;
+
+			document.body.addEventListener("click", function () {
+				// always clear menu icon hover state
+				self.setOverflowMenuIconHover(false);
+				self.services.events.dispatchEvent(
+					Events.Toolbar.HIDE_MENU_ITEMS
+				);
+			});
+
+			if (hasOpenedOverflowMenuOptions) {
+				this.setOverflowMenuIconHover(true);
+				this.overflowMenuOptions.html(this.getOverflowMenuHTML());
+			} else {
+				this.setOverflowMenuIconHover(false);
+				this.overflowMenuOptions.html(null);
+			}
+		}
+	}
+
+	getOverflowMenuHTML() {
+		// supports only reset zoom for now
+		// if getResetZoomMenuItem() return nothing
+		// don't render whole overflow menu
+		if (!this.getResetZoomMenuItem()) {
+			return "";
+		}
+
+		let defaultHTML;
+		defaultHTML = `<div data-floating-menu-container="true"
+			data-floating-menu-direction="bottom" role="main">
+			<ul class="bx--overflow-menu-options bx--overflow-menu--flip bx--overflow-menu-options--open"
+				tabindex="-1" role="menu" aria-label="Menu" data-floating-menu-direction="bottom"
+				style="left:${this.overflowMenuX}px; top:${this.overflowMenuY}px;">`;
+		defaultHTML += this.getResetZoomMenuItem();
+		defaultHTML += `</ul></div>`;
+
+		return defaultHTML;
+	}
+
+	getResetZoomMenuItem() {
```

Comment:
`getDropdownItems()`?

Reply:
This is specific to `Reset zoom` menu item, not for all items.